File: DesignPatternsPractices/DesignPatternsPractices/DesignPatterns/Command/CodeTemplate/Command.m
//
//  Command.m
//  DesignPatternsPractices
//
//  Created by Su XinDe on 14-9-6.
//  Copyright (c) 2014年 Su XinDe. All rights reserved.
//

#import "Command.h"

@implementation Command

- (id)initWithReceiver:(Receiver *)receiver
{
    if (self = [super init]) {
        mReceiver = [receiver retain];
    }
    return self;
}

- (void)execute {}

@end

File: Others/DesignPattern/src/Command/Command.java
package Command;

/**
 * 声明执行操作的接口
 * @author Administrator
 *
 */
public abstract class Command {
    
	//知道如何实现与执行一个请求相关的操作。任何类都可能作为一个接收者。
    protected Receiver receiver;
    
    public Command(Receiver receiver) {
        this.receiver = receiver;
    }
    
    public abstract void execute();
}

